### PR TITLE
Remove runFirstColumn for top level aggregate where

### PR DESCRIPTION
**Pull request by Liam-Doodson**
# Description

Removed `runFirstColumn` usage from top-level aggregate where. For example:

```graphql
mutation {
    updateUsers(
        where: { likedPostsAggregate: { count: 2 } } # This no longer uses runFirstColumn
        update: { 
            likedPosts: {
                where: { 
                    node: {
                        likesAggregate: {
                            count: 2 # This would still use runFirstColum
                        }
                    } 
                } 
                update: {
                    node: {
                        content: "New Content"
                    } 
                } 
            } 
    }) {
        users {
            name
            likedPosts {
                id
                content
            }
        }
    }
}
```

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2472.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/src/translate/translate-top-level-match.ts [hunk 1]
```diff
@@ -122,23 +156,79 @@ function createFulltextMatchClause(
...
+
+    return {
+        matchClause,
+        whereOperators,
+    };
+}
 
-    return matchQuery;
+export function preComputedWhereFields(
+    whereInput: any,
```
Comment (angrykoala) [inline packages/graphql/src/translate/translate-top-level-match.ts:185]: Can we get a bit more accurate type here? Like `Record<string, any>`?

**Review by angrykoala — APPROVED** [review]
Some minor proposals of code improvements

File: packages/graphql/src/translate/create-aggregate-where-and-params.ts [hunk 2]
```diff
@@ -17,14 +17,189 @@
...
+                    .map((whereInput) => {
+                        return aggregateWhere(whereInput, refNode, aggregationTarget, cypherRelation);
+                    })
+                    .reduce(aggregateWhereReducer, {
+                        returnVariables: [],
+                        predicates: [],
+                    });
+
+                return {
+                    returnVariables,
```
[the remark this answers is not in the document]
Comment (Liam-Doodson) [reply packages/graphql/src/translate/create-aggregate-where-and-params.ts]: Sorry, I don't think I understand what you mean here. Can you clarify?

**Pull request merged**